### PR TITLE
fix warpstream integration tests

### PR DESCRIPTION
**Pull request by shouples**
## Summary of Changes

Swaps the WarpStream agent Docker image tag from `latest` ([v822](https://docs.warpstream.com/warpstream/overview/change-log#release-v822)) to [`v821`](https://docs.warpstream.com/warpstream/overview/change-log#release-v821) to fix currently-breaking integration tests on `main` ([example](https://semaphore.ci.confluent.io/jobs/757f88c3-6f51-4db9-be83-7752562cea4e/summary?report_id=17990af8-cb17-371c-9a8e-215e0e201902)).  

> [!NOTE]
> [v823](https://docs.warpstream.com/warpstream/overview/change-log#release-v823) was released today to include a fix for the error below, but pinning to a version is still a good defensive measure.

This container issue was also seen from `docker run public.ecr.aws/warpstream-labs/warpstream_agent:latest demo` (when `latest` was still v822):
```
Starting local agents...
Setting up the demo environment, this can take up to 10s...
panic: error starting tableflow agent: failed to initialize duckdb client dependencies: failed to initialize embedded duckdb cli: duckdb process failed: exit status 127 (stderr: /tmp/warpstream-duckdb/duckdb: error while loading shared libraries: libstdc++.so.6: cannot open shared object file: No such file or directory
	)

goroutine 28 [running]:
github.com/warpstreamlabs/warpstream/pkg/logging.Fatalf(...)
	/opt/actions-runner/_work/warpstream/warpstream/pkg/logging/logging.go:50
github.com/warpstreamlabs/warpstream/pkg/logging.FatalfCLI(...)
	/opt/actions-runner/_work/warpstream/warpstream/pkg/logging/logging.go:58
main.runPlayground.func7()
	/opt/actions-runner/_work/warpstream/warpstream/services/agent/app/playground.go:505 +0x19c
created by github.com/warpstreamlabs/warpstream/pkg/xgo.GoWithoutRecover in goroutine 1
	/opt/actions-runner/_work/warpstream/warpstream/pkg/xgo/xgo.go:6 +0x1c
```

Dropping to `v821` allows the container to start as intended, and tests are now passing.

## Pull request checklist

Please check if your PR fulfills the following (if applicable):

- Tests:
    - [ ] Added new
    - [ ] Updated existing
    - [ ] Deleted existing
- [ ] Have you validated this change locally against a running instance of the Quarkus dev server?
    ```shell
    make quarkus-dev
    ```
- [ ] Have you validated this change against a locally running native executable?
    ```shell
    make mvn-package-native && ./target/ide-sidecar-*-runner
    ```

### CODE REVIEW

File: src/test/java/io/confluent/idesidecar/restapi/integration/WarpStreamRegressionIT.java [hunk 1]
```diff
@@ -16,11 +16,12 @@
...
-      // `DescribeCluster.controller()` response. The sidecar's Internal Kafka REST
-      // previously failed to handle this case correctly. This test class ensures that
-      // the ClusterV3Suite passes since the sidecar now correctly handles this case.
-      "latest"
+      // Pin the stable channel rather than `latest`: `latest` is a moving tag that tracks the
+      // newest (possibly pre-release) agent, so a republish can break CI on every branch at once.
+      // This suite verifies that the sidecar's Internal Kafka REST (the ClusterV3Suite) works
+      // against WarpStream; it originally regression-guarded a version that returned `null` for
+      // `DescribeCluster.controller()`, an edge case newer stable builds may no longer exercise.
+      "latest-stable"
```
[the remark this answers is not in the document]
Comment (shouples) [reply src/test/java/io/confluent/idesidecar/restapi/integration/WarpStreamRegressionIT.java]: Pinning to the last stable version in https://github.com/confluentinc/ide-sidecar/pull/549/commits/a515addda89c27972cead68c730af92bd09684e0; new test in https://github.com/confluentinc/ide-sidecar/pull/549/commits/27492d0c4cdf96b5e08a026ecd2d098704e1ffb1

**Review by jlrobins — APPROVED** [review]
